pacific: mgr/pg_autoscaler: Introduce autoscaler scale-down feature (#42428)
ceph/ceph: doc/rados/operations/placement-groups.rst

to OSDs of class `hdd` will each have optimal PG counts that depend on
the number of those respective device types.

> As described above, the autoscaler by default starts out each pool with 
> minimal pgs and `scale-up` the pgs when there is more usage in each pool.
> However, it also has the option to switch to a `scale-down` profile
> where each pool starts out with a full complements of pgs and only 
> `scale-down` when the usage ratio across the pools is not even.

Reviewer: ```suggestion
The autoscaler uses the `scale-up` profile by default, where it starts out each pool with minimal PGs and scales up PGs when there is more usage in each pool.
However, it also has a `scale-down` profile,
where each pool starts out with a full complements of PGs and only 
scales down when the usage ratio across the pools is not even.
```

> With only the `scale-down` profile, the autosacler identifies

Reviewer: ```suggestion
With the `scale-down` profile, the autoscaler identifies
```

any overlapping roots and prevents the pools with such roots

> from scaling. Since, the overlapping roots can cause problems

Reviewer: ```suggestion
from scaling because overlapping roots can cause problems
```

with the scaling process.

> executed with the command::
> 
>   osd pool set autoscale-profile scale-down
> 
> Respectively we can switch back to the default `scale-up` profile
> by executing the command::
> 
>   osd pool set autoscale-profile scale-up

Reviewer: ```suggestion
To use the `scale-down` profile::

  ceph osd pool set autoscale-profile scale-down

To switch back to the default `scale-up` profile::

  ceph osd pool set autoscale-profile scale-up
```
